include/zephyr/bluetooth/bluetooth.h

```diff
@@ -347,9 +347,9 @@ bool bt_is_ready(void);
  *
  * Set Bluetooth GAP Device Name.
  *
- * When advertising with device name in the advertising data the name should
- * be updated by calling @ref bt_le_adv_update_data or
- * @ref bt_le_ext_adv_set_data.
+ * @note The advertising data is not automatically updated. When advertising with device name in the
+ * advertising data, the name should be updated by calling @ref bt_le_adv_update_data or
+ * @ref bt_le_ext_adv_set_data after the call to this function.
```

Comment:
Interesting note: We don't have a callback when the device name is changed, so if multiple applications exists on the device, they may "know" different device names